internal/controller/rotators/gcp_oidc_token_rotator_test.go

```diff
@@ -871,34 +874,80 @@ func TestGetGCPProxyClientOption(t *testing.T) {
 				}
 			}()
 
-			if tt.setEnvVar {
+			if tt.setProxyEnv {
 				os.Setenv("AI_GATEWAY_GCP_AUTH_PROXY_URL", tt.proxyURL)
 			} else {
 				os.Unsetenv("AI_GATEWAY_GCP_AUTH_PROXY_URL")
 			}
 
 			// Call the function under test.
-			got, err := getGCPProxyClientOption()
+			roundTripper, err := newBearerAuthRoundTripper(tt.token)
 
 			// Validate error expectation.
 			if tt.wantErr {
 				require.Error(t, err)
-				require.Contains(t, err.Error(), "invalid proxy URL")
+				require.Nil(t, roundTripper)
 				return
 			}
 
 			require.NoError(t, err)
+			require.NotNil(t, roundTripper)
 
-			// Validate nil option expectation.
-			if tt.wantNilOption {
-				require.Nil(t, got)
-				return
-			}
+			// Verify it's the expected type.
+			bearerRT, ok := roundTripper.(*bearerAuthRoundTripper)
+			require.True(t, ok, "Expected bearerAuthRoundTripper type")
+			require.Equal(t, tt.token, bearerRT.token)
+			require.NotNil(t, bearerRT.base)
+		})
+	}
+}
 
-			// Additional validation if provided.
-			if tt.validateOption != nil {
-				tt.validateOption(t, got)
-			}
+func TestBearerAuthRoundTripper_RoundTrip(t *testing.T) {
+	tests := []struct {
+		name             string
+		token            string
+		expectAuthHeader bool
+	}{
+		{
+			name:             "with token",
+			token:            "test-bearer-token",
+			expectAuthHeader: true,
+		},
+		{
+			name:             "without token",
+			token:            "",
+			expectAuthHeader: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create a test server to capture the request.
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				authHeader := r.Header.Get("Authorization")
+				if tt.expectAuthHeader {
+					expectedAuth := "Bearer " + tt.token
+					require.Equal(t, expectedAuth, authHeader, "Authorization header mismatch")
+				} else {
+					require.Empty(t, authHeader, "Authorization header should be empty")
+				}
+				w.WriteHeader(http.StatusOK)
+			}))
+			defer server.Close()
+
+			// Create the round tripper.
+			roundTripper, err := newBearerAuthRoundTripper(tt.token)
+			require.NoError(t, err)
+
+			// Create a request to the test server.
+			req, err := http.NewRequest(http.MethodGet, server.URL, nil)
+			require.NoError(t, err)
+
+			// Execute the request through the round tripper.
+			resp, err := roundTripper.RoundTrip(req)
+			require.NoError(t, err)
+			require.Equal(t, http.StatusOK, resp.StatusCode)
+			resp.Body.Close()
```

Comment:
can you do defer instead of inline? this won't be called if the assertion failed in the middle.